API_request_test.py

```diff
@@ -12,16 +46,24 @@
 # https://www.weatherapi.com/signup.aspx
 # The url is set to get forecast data. For historical data it should be changed
 
+# KeyError exception handling
 url_wheatherapi = 'https://api.weatherapi.com/v1/forecast.json'
-API_KEY_wheatherapi = <paste here your personal API key>
+try:
+    API_KEY_wheatherapi = keys_dict['wheatherapi.com']
+except KeyError:
+    print("The key name in the file better_weather_keys.txt is incorrect, should be 'wheatherapi.com'")
 
 # You can change the name of the city <q parameter>, search engine is quite efficient
 # days=1 means that you get hourly forecast for one date, the day you are make a request. 
 # To get forecast for next day you need additional parameter
-params_wheatherapi = dict(key=API_KEY_wheatherapi, q='Przemysl', days=1) 
+params_wheatherapi = dict(key=API_KEY_wheatherapi, q=args.location, days=1) 
 
 resonse_wheatherapi = requests.get(url_wheatherapi, params=params_wheatherapi)
 
+# check if lack of proper request response is due to 403 error
+if resonse_wheatherapi.status_code == 403:
+    print("Supplied API key for wheatherapi.com is wrong.")
+
 # print(resonse_wheatherapi)
 # print(resonse_wheatherapi.status_code)
 # print(resonse_wheatherapi.headers)
```

Comment:
All these commented-out prints are left on purpose or can we just remove them? This question applies also to lines 92-94, 124-132, and 170-172.

Reply:
Yes. I know that those comments give no additional info, but they might be handy in the future and I want to keep them in this POC. In the proper program components I will remove all not important comments.

Reply:
You can always keep such extra info on a separate dev branch to not spoil the main code base. But ok, let's leave it as it is for now.